flake.nix

```
@@ -16,9 +16,9 @@
         # Add the commit to the version string, in case someone builds from main
         getVersion = pkgs.lib.trivial.pipe self [
           (x: "${lastTag}")
-          (x: if (self ? revCount)
+          (x: if (self ? shortRev)
               then "${x}-${self.shortRev}"
-              else "${x}-${self.dirtyShortRev}")
+              else "${x}-${if self ? dirtyShortRev then self.dirtyShortRev else "dirty"}")
```

There's also an [or syntax] that provides an alternative value when an attribute doesn't exist.

```suggestion
              else "${x}-${self.dirtyShortRev or "dirty"}")
```

[or syntax]: https://nix.dev/manual/nix/2.24/language/operators#attribute-selection